File: divide_and_conquer/starter_code/convoy.py
n, k = map(int, input().split())

speeds = [] # driver i can travel to the stadium in speeds[i] seconds
for _ in range(n):
    speeds.append(int(input()))
speeds.sort()

def can_drive(time):
    # TODO: if can drive in time, return true. Otherwise return false.
    return True


hi = ?? # TODO: what is upper bound?
lo = ?? # TODO: what is lower bound?

# TODO: bisection method here

print(hi)

File: divide_and_conquer/starter_code/convoy.cpp
#include <stdio.h>
#include <algorithm>

using namespace std;

int64_t n, k, speeds[20000];

bool can_drive(int64_t time) {
    // TODO: if can drive in time, return true. Otherwise return false.
    return true;
}

int main() {
    scanf("%ld %ld", &n, &k);
    for (int i = 0; i < n; i++) scanf("%ld", speeds + i);
    sort(speeds, speeds + n - 1);

    int64_t hi = ??; // TODO: what is upper bound? 
    int64_t lo = ??; // TODO: what is lower bound?
    int64_t mid; 

    // TODO: bisection method here

    printf("%ld\n", hi);
    return 0;
}
